src/main/java/eu/ai4work/sws/controller/SlidingDecisionController.java

```diff
@@ -22,19 +22,49 @@ public class SlidingDecisionController {
     private final SlidingDecisionService slidingDecisionService;
     private final ApplicationScenarioConfiguration applicationScenarioConfiguration;
 
+    /**
+     * Processes a sliding decision request by validating the input parameters from the sliding decision request,
+     * executing the decision logic and returns a decision response.
+     *
+     * @param request The request body containing input parameters for decision process
+     * @return SlidingDecisionResponse containing decision status and decision details
+     */
     @PostMapping("/sliding-decision")
     public SlidingDecisionResponse processSlidingDecisionRequest(@RequestBody SlidingDecisionRequest request) throws Exception {
-        SlidingDecisionResponse response = new SlidingDecisionResponse();
-        response.setDecisionStatus(SlidingDecisionStatus.RESPONSE);
+        validateSlidingDecisionInputParameters(request.getSlidingDecisionInputParameters());
 
         SlidingDecisionResult decisionResult = slidingDecisionService.getSlidingDecision(request.getSlidingDecisionInputParameters());
 
+        return createResponse(decisionResult);
+    }
+
+    /**
+     * Creates a sliding decision response containing decision status and decision result details.
+     * Decision result details include the decision status and description.
+     *
+     * @param decisionResult Evaluated sliding decision result after applying the decision rules
+     * @return SlidingDecisionResponse containing decision status and decision details
+     */
+    private SlidingDecisionResponse createResponse(SlidingDecisionResult decisionResult) {
         Map<String, Object> decisionResultDetails = new HashMap<>();
         decisionResultDetails.put(SLIDING_DECISION, decisionResult);
-        decisionResultDetails.put(DESCRIPTION,  applicationScenarioConfiguration.getDecisionResultsDescription().get(decisionResult.name()));
+        decisionResultDetails.put(DESCRIPTION, applicationScenarioConfiguration.getDecisionResultsDescription().get(decisionResult.name()));
 
-        response.setDecisionResult(decisionResultDetails);
+        return SlidingDecisionResponse.builder()
+                .decisionStatus(SlidingDecisionStatus.RESPONSE)
+                .decisionResult(decisionResultDetails)
+                .build();
+    }
 
-        return response;
+    /**
+     * Validates the input parameters from the sliding decision request.
+     *
+     * @param slidingDecisionInputParameters The input parameters from the sliding decision request
+     * @throws IllegalArgumentException if parameter map is null or empty
+     */
+    private void validateSlidingDecisionInputParameters(Map<String, Object> slidingDecisionInputParameters) {
```

Comment:
Wondering about the name: it does not fully validate the parameters (other checks are done in the rule engine), but just assures they are not empty. 

Maybe a more precise name would be:
```suggestion
    private void assureInputParametersAreNotEmpty(Map<String, Object> slidingDecisionInputParameters) {
```